R/create_protocol.R

```diff
@@ -397,23 +387,86 @@ get_short_titles <- function(
-                    pattern = "^(.*)\\/(.*)(\\.Rmd)$",
-                    replacement = "\\2")
-  lf <- str_subset(string = lf,
+  ld <- str_subset(string = ld,
                    pattern = protocol_type)
-  lf <- str_subset(string = lf,
+  ld <- str_subset(string = ld,
                    pattern = paste0("(", language, ")$"))
-  lf <- str_extract(string = lf,
-                    pattern = "(?<=\\d{3}_).*")
-  lf <- lf[!is.na(lf)]
+  ld <- str_extract(string = ld,
+                    pattern = "(?<=\\d{3}_)([a-z]|-)*")
+  ld <- ld[!is.na(ld)]
+
+  return(ld)
+}
+
+#' @title Internal function to get (or set) the full path to a protocol.
+#'
+#' @param protocol_folder_name Character string giving the name of the protocol subfolder
+#' @param theme A character string equal to one of `"generic"`,
+#' `"water"`, `"air"`, `"soil"`, `"vegetation"` or `"species"`.
+#' Defaults to NULL.
+#' Only needed if no folder with the name of the protocol subfolder exists and the request is for a thematic protocol.
+#' @param project Character string giving the name of the project folder.
+#' Defaults to NULL.
+#' Only needed if no folder with the name of the protocol subfolder exists and the request if for a project-specific protocol.
+#'
+#' @return A character vector containing the full path to the protocol.
+#'
+#' @importFrom rprojroot find_root is_git_root
+#' @importFrom assertthat assert_that is.string
+#' @importFrom stringr str_subset
+#'
+#' @export
+#'
+#'
+#' @examples
+#' \dontrun{
+#' get_path_to_protocol(protocol_folder_name = "sfp_401-vegopname-terrest_nl")
+#'}
+get_path_to_protocol <- function(protocol_folder_name = NULL,
+                                 theme = NULL,
+                                 project = NULL) {
+  assert_that(is.string(protocol_folder_name))
+
+  # first case: the path exists already
+  project_root <- find_root(is_git_root)
+  ld <- list.dirs(path = file.path(project_root,"src"),
+                  full.names = TRUE,
+                  recursive = TRUE)
+  ld <- str_subset(string = ld,
+                   pattern = protocol_folder_name)
+  if (!identical(ld, character(0))) {
+    path_to_protocol <- ld[[1]]
+  } else {
+    # second case: the path does not yet exist
+    if (is.null(theme) & is.null(project) |
+        is.string(theme) & is.string(project)) {
+      stop("Check the spelling of protocol_folder_name - or - provide a string value for theme or project, not both.")
+    }
+
+    if (is.string(theme)) {
+      subfolder_of <- "thematic"
+      protocol_leading_number <- themes_df[themes_df$theme == theme,
+                                           "theme_number"]
+      theme <- paste0(protocol_leading_number, "_", theme)
+      path_to_protocol <- file.path(project_root,
+                                    "src",
+                                    subfolder_of,
+                                    theme,
+                                    protocol_folder_name)
+    } else {
```

Comment:
you can use an early return in this case. That make longer if else statements more reable.

```
if(condition) {
    # do something when TRUE
    return(output) # returns output and ends the function
}
# do something else. You only get here when condition == FALSE
return(output)
```

Reply:
See 8131735